src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -303,19 +305,19 @@ public boolean isEnabled() {
 	    return true;
     }
     public void initEncoders() {
-
+    	trySetSparkMaxPosition(frontLeftEncoder,0);
+    	trySetSparkMaxPosition(frontLeftEncoder,0);
+    	trySetSparkMaxPosition(rearLeftEncoder,0);
+    	trySetSparkMaxPosition(frontRightEncoder,0);
+    	trySetSparkMaxPosition(rearRightEncoder,0);  
         referenceAvgPosition = getAveragePosition();
             	
```

Comment:
I don't like getAveragePosition being called immediately after an encoder reset.  I just don't trust that the values read from the encoders will be up to date immediately.  Since initEncoders is only called at construction, I'd rather see us set referenceAvgPosition explicitly to 0.0 at this point.  Most of the commands call resetEncoders() in their initialize() method anyway and will get the correct reference value for the encoder revolutions at that point.

Reply:
OH you are right! EUREKA! this could be why this solution didnt work at the shop when we tried it